target_chains/sui/contracts/sources/pyth_accumulator.move

```diff
@@ -0,0 +1,157 @@
+    public fun test_get_price_feed_updates_from_accumulator(accumulator_message: vector<u8>, wormhole_state: &WormState, clock: &Clock): vector<PriceInfo> {
+        let _PYTHNET_ACCUMULATOR_UPDATE_MAGIC = 1347305813;
+        let cur = cursor::new(accumulator_message);
+        let header: u32 = deserialize::deserialize_u32(&mut cur);
+        let updates = if ((header as u64) == _PYTHNET_ACCUMULATOR_UPDATE_MAGIC) {
+            parse_and_verify_accumulator_message_with_worm_state(&mut cur, wormhole_state, clock)
+        }
+        else {
+            // don't handle this code path here
+            abort 99
+        };
+        //TODO -  update_cache(updates);
+        // cursor should be empty after parsing the header and accumulator message
+        cursor::destroy_empty(cur);
+        updates
+    }
+
+    #[test_only]
+    /// Given a cursor at the beginning of an accumulator message, verifies the validity of the message and the
+    /// embedded VAA, parses and verifies the price updates and returns an array of PriceInfo representing the updates
+    /// Note: this function is adapted from the Aptos pyth accumulator
+    /// Note: this function is test-only because it is not used in prod, since the verified VAA would be passed into this function.
+    ///       We don't verify the VAA inside of there, because in that case the wormhole package effectively becomes hard-coded,
+    ///       so when the WH package upgrades, this code will be broken.
+    public fun parse_and_verify_accumulator_message_with_worm_state(cursor: &mut Cursor<u8>, wormhole_state: &WormState, clock: &Clock): vector<PriceInfo> {
+        let major = deserialize::deserialize_u8(cursor);
+        assert!(major == 1, E_INVALID_ACCUMULATOR_PAYLOAD);
+        let _minor = deserialize::deserialize_u8(cursor);
+
+        let trailing_size = deserialize::deserialize_u8(cursor);
+        deserialize::deserialize_vector(cursor, (trailing_size as u64));
+
+        let proof_type = deserialize::deserialize_u8(cursor);
+        assert!(proof_type == 0, E_INVALID_ACCUMULATOR_PAYLOAD);
+
+        let vaa_size = deserialize::deserialize_u16(cursor);
+        let vaa = deserialize::deserialize_vector(cursor, (vaa_size as u64));
+        let msg_vaa = vaa::parse_and_verify(wormhole_state, vaa, clock);
+        // TODO - verify VAA targets this chain and contract
+        //verify_data_source(&msg_vaa);
+        let merkle_root_hash = parse_accumulator_merkle_root_from_vaa_payload(vaa::payload(&msg_vaa));
+        vaa::destroy(msg_vaa);
+        parse_and_verify_accumulator_updates(cursor, merkle_root_hash, clock)
+    }
+
+    // parse_and_verify_accumulator_message verifies that the merkle root contained a pre-verified VAA
+    // matches the rest of the content in the accumulator message (encapsulated by cursor).
+    public(friend) fun parse_and_verify_accumulator_message(cursor: &mut Cursor<u8>, vaa_payload: vector<u8>, clock: &Clock): vector<PriceInfo> {
```

Comment:
We don't need both `parse_and_verify_accumulator_message` and `parse_and_verify_accumulator_message_with_worm_state` in the end right?

Reply:
yeah, only the former is needed, the latter is marked `#[test_only]`

Reply:
I'm confused about why we need the `_with_worm_state` version of this method for the tests -- seems like we should be testing the code paths in this function here, and not the ones above (which could be different). I think a better test helper would parse the vaa_payload out of the accumulator message so that the tests could call this function.

Reply:
the `_with_worm_state` version of the function is a shortcut used for testing `pyth_accumulator.move` and not the full code path in `pyth.move`. I had to move those tests to pyth.move, because of a dependency cyckle.

As noted in pyth.move, 
`// pyth accumulator tests (included in this file instead of pyth_accumulator.move to avoid dependency cycle - as we need pyth_tests::setup_test)`

Reply:
Specifically, `test_get_price_feed_updates_from_accumulator` is used in `test_parse_and_verify_accumulator_updates` in `pyth.move`.

Reply:
that test function isn't for testing the full code path, it's merely a shortcut for testing accumulator-specific functions, such as `parse_accumulator_merkle_root_from_vaa_payload`.

Reply:
good idea, I removed the `parse_and_verify_accumulator_message_with_worm_state` entirely, and instead added the helper `parse_vaa_bytes_from_accumulator_message` as you suggested.